src/main/java/de/tum/in/www1/artemis/config/localvcci/ssh/SshConfiguration.java

```diff
@@ -0,0 +1,87 @@
+import org.apache.sshd.server.SshServer;
+import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+import de.tum.in.www1.artemis.config.localvcci.ssh.service.GitPublickeyAuthenticatorService;
+import de.tum.in.www1.artemis.config.localvcci.ssh.service.SshGitCommandFactoryService;
+import de.tum.in.www1.artemis.config.localvcci.ssh.service.SshGitLocationResolverService;
+
+@Profile(PROFILE_LOCALVC)
+@Configuration
+public class SshConfiguration {
+
+    private static final Logger log = LoggerFactory.getLogger(SshConfiguration.class);
+
+    @Value("${artemis.version-control.ssh-port:7921}")
+    private int sshPort;
+
+    @Value("${artemis.version-control.ssh-host-key-path:null}")
+    private Optional<Path> sshHostKeyPath;
+
+    @Value("${server.url}")
+    private String artemisServerUrl;
+
+    private final GitPublickeyAuthenticatorService gitPublickeyAuthenticatorService;
+
+    private final SshGitCommandFactoryService sshGitCommandFactoryService;
+
+    private final SshGitLocationResolverService sshGitLocationResolverService;
+
+    public SshConfiguration(GitPublickeyAuthenticatorService gitPublickeyAuthenticatorService, SshGitCommandFactoryService sshGitCommandFactoryService,
+            SshGitLocationResolverService sshGitLocationResolverService) {
+        this.gitPublickeyAuthenticatorService = gitPublickeyAuthenticatorService;
+        this.sshGitCommandFactoryService = sshGitCommandFactoryService;
+        this.sshGitLocationResolverService = sshGitLocationResolverService;
+    }
+
+    /**
+     * Configure and start the SSH server for LocalVC
+     *
+     * @return the configured SSH server
+     */
+    @Bean(initMethod = "start", destroyMethod = "stop")
+    public SshServer sshServer() {
+        SshServer sshd = SshServer.setUpDefaultServer();
+        sshd.setPort(sshPort);
+
+        if (sshHostKeyPath.isPresent() && sshHostKeyPath.get().toFile().exists()) {
+            // this allows to customize the host key file for production environments
+            log.info("Use SSH host key {}", sshHostKeyPath.get());
+            sshd.setKeyPairProvider(new MultipleHostKeyProvider(sshHostKeyPath.get()));
+        }
+        else {
+            // this is a simple solution for development, the host key will be generated during the first ssh operation in case it does not exist
+            sshd.setKeyPairProvider(new SimpleGeneratorHostKeyProvider(Paths.get("tmp", "hostkey.ser")));
+        }
+        sshd.setCommandFactory(
+                sshGitCommandFactoryService.withGitLocationResolver(sshGitLocationResolverService).withExecutorServiceProvider(() -> ThreadUtils.newFixedThreadPool("git-ssh", 8)));
```

Comment:
Why do we use a fixed thread pool of exactly size 8? Should we use a CachedThreadPool to make this more dynamic?

Reply:
Tbh I don't know. I'll keep this in mind and try it out in a follow-up PR